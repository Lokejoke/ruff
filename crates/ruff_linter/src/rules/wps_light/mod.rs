//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
