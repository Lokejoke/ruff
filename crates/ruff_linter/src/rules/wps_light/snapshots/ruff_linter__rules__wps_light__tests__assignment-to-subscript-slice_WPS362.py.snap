---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
snapshot_kind: text
---
WPS362.py:1:1: WPS362 Assignment to a subscript slice found
  |
1 | a[1:3] = [1, 2]
  | ^^^^^^ WPS362
2 | a[slice(1)] = [1, 3]
  |

WPS362.py:4:1: WPS362 Assignment to a subscript slice found
  |
2 | a[slice(1)] = [1, 3]
3 | 
4 | a[:0], a[-1:] = prefix, suffix
  | ^^^^^ WPS362
5 | 
6 | a[:] = complete_replacement
  |

WPS362.py:4:8: WPS362 Assignment to a subscript slice found
  |
2 | a[slice(1)] = [1, 3]
3 | 
4 | a[:0], a[-1:] = prefix, suffix
  |        ^^^^^^ WPS362
5 | 
6 | a[:] = complete_replacement
  |

WPS362.py:6:1: WPS362 Assignment to a subscript slice found
  |
4 | a[:0], a[-1:] = prefix, suffix
5 | 
6 | a[:] = complete_replacement
  | ^^^^ WPS362
7 | 
8 | a[0:0] = prepend
  |

WPS362.py:8:1: WPS362 Assignment to a subscript slice found
   |
 6 | a[:] = complete_replacement
 7 | 
 8 | a[0:0] = prepend
   | ^^^^^^ WPS362
 9 | 
10 | a[0] = 1
   |

WPS362.py:16:1: WPS362 Assignment to a subscript slice found
   |
14 | board = [["_"] * size for _ in range(size)]
15 | 
16 | board[0:1][1:2] = "X"
   | ^^^^^^^^^^^^^^^ WPS362
17 | 
18 | board[0] = board[0:1] = board[1:2] = [board[0][:], board[1][:], board[2][:]] = board
   |

WPS362.py:18:12: WPS362 Assignment to a subscript slice found
   |
16 | board[0:1][1:2] = "X"
17 | 
18 | board[0] = board[0:1] = board[1:2] = [board[0][:], board[1][:], board[2][:]] = board
   |            ^^^^^^^^^^ WPS362
   |

WPS362.py:18:25: WPS362 Assignment to a subscript slice found
   |
16 | board[0:1][1:2] = "X"
17 | 
18 | board[0] = board[0:1] = board[1:2] = [board[0][:], board[1][:], board[2][:]] = board
   |                         ^^^^^^^^^^ WPS362
   |

WPS362.py:18:39: WPS362 Assignment to a subscript slice found
   |
16 | board[0:1][1:2] = "X"
17 | 
18 | board[0] = board[0:1] = board[1:2] = [board[0][:], board[1][:], board[2][:]] = board
   |                                       ^^^^^^^^^^^ WPS362
   |

WPS362.py:18:52: WPS362 Assignment to a subscript slice found
   |
16 | board[0:1][1:2] = "X"
17 | 
18 | board[0] = board[0:1] = board[1:2] = [board[0][:], board[1][:], board[2][:]] = board
   |                                                    ^^^^^^^^^^^ WPS362
   |

WPS362.py:18:65: WPS362 Assignment to a subscript slice found
   |
16 | board[0:1][1:2] = "X"
17 | 
18 | board[0] = board[0:1] = board[1:2] = [board[0][:], board[1][:], board[2][:]] = board
   |                                                                 ^^^^^^^^^^^ WPS362
   |
